eva_model/cluster_images.py

```diff
@@ -0,0 +1,230 @@
+import io
+import json
+import logging
+import os
+import random
+from urllib.parse import urlparse
+import requests
+import asyncio
+
+import boto3
+import cv2
+import nest_asyncio
+from botocore.exceptions import ClientError
+from eva.server.db_api import connect
+from evalabeling.utils import DATA_UNDEFINED_NAME
+from evalabeling.model import EvaLabelingBase
+from label_studio_tools.core.utils.io import get_data_dir
+
+logger = logging.getLogger(__name__)
+
+EVA_CURSOR = connect(host='127.0.0.1', port=5432).cursor()
+
+# TODO: Remove hardcoded url and key
+
+MAIN_URL = "http://localhost:8080"
+API_KEY = 'e5234515ff4ed7f6281e6888ce0c5ee4967389f6'
+
+def json_load(file, int_keys=False):
+    with io.open(file, encoding='utf8') as f:
+        data = json.load(f)
+        if int_keys:
+            return {int(k): v for k, v in data.items()}
+        else:
+            return data
+
+def remove_all_predictions():
+    url = f"{MAIN_URL}/api/predictions/"
+    headers = {
+        'Authorization': f'Token {API_KEY}' 
+    }
+    r = requests.get(url=url, headers=headers, data={})
+    r_obj = json.loads(r.text)
+    if not type(r_obj)== dict:
+        for it in r_obj:
+            url = f"{MAIN_URL}/api/predictions/{it['id']}"
+            r = requests.delete(url=url, headers=headers, data={})
+            if not r.status_code==204:
+                print("error deleting file:", url, headers)
+
+image_for_similarity = None
+
+class EVAModel(EvaLabelingBase):
+    """
+    EVA connection using Label Studio ML backend server. This will allow you to run EVA queries on Label Studio.
+    """
+
+    EVA_CURSOR = None
+
+    def __init__(self, image_dir=None, labels_file=None, score_threshold=0.3, device='cuda', **kwargs):
```

Comment:
Is this threshold something to be documented?

Reply:
This is part of the Label Studio ML template